cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -92,13 +92,23 @@ void VhpiImpl::get_sim_time(uint32_t *high, uint32_t *low)
     *low = vhpi_time_s.low;
 }
 
+template <typename T>
+int log10int(T v)
+{
+    int i = -1;
+    do {
+        v /= 10;
+        i += 1;
+    } while (v);
+    return i;
+}
```

Comment:
Proof that this is a) correct, and b) fixes a bug:

https://godbolt.org/z/jcgrzN